### ISSUE TITLE
not updating ArgoCD itself...

### ISSUE THREAD

**Issue by reefland** [mjpitz/renovate-gitops#3]
Greetings,  after reading your "Renovate your GitOps" blog post, deployed it.   Overall it works really well, renovate is updating many applications.  But not ArgoCD itself.

Renovate generates the PR.  I process it on GitHub, all no issues.   Log into ArgoCD it syncs. It shows the sync state with the PR commit in the status comment.  ArgoCD Status is syncd.  But nothing happens.  The increment in the chart version doesn't seem to trickle down to updating image tags and redeploying pods with newer version.

For example, renovate just updated the `workloads/argocd/Chart.yaml` to `4.8.0`:
```
apiVersion: v2
type: application
name: argocd
version: 0.0.0 # unused
appVersion: 0.0.0 # unused
dependencies:
  - name: argo-cd
    repository: https://argoproj.github.io/argo-helm
    version: 4.8.0
    condition: argo-cd.enabled
```

Inspecting the pods, they still show older version of ArgoCD:
```
app.kubernetes.io/name: argocd-server
    app.kubernetes.io/part-of: argocd
    app.kubernetes.io/version: v2.3.3
    helm.sh/chart: argo-cd-4.6.0
```

If I run renovate again, it thinks ArgoCD is up to date.  No new PR is created.

**Comment by mjpitz** [comment]
Is there any additional information about synchronization errors in the ArgoCD console or available via CLI?

**Comment by reefland** [comment]
Thank you for the prompt response... I figured it out.  

1. I initially disabled selfHeal to prevent ArgoCD from resetting admin password to default.
2. I tried adding an ignoreDifferences to the ArgoCD secret to prevent resetting password. Sort of worked, until I tried a manual sync/refresh it reset the password back to the default.
3. I finally disable secret generation and create an external secret. This is working perfectly.

But I did not re-enable the selfHeal on ArgoCD.  I has assumed I could manually sycn / replace / prune / refresh to trigger an upgrade.  

Once I enabled selfHeal on ArgoCD, the upgrade worked as expected.   Closing issue.
